be/src/exec/pipeline/crossjoin/cross_join_left_operator.h

```diff
@@ -153,29 +156,30 @@ class CrossJoinLeftOperatorFactory final : public OperatorWithDependencyFactory
 public:
     CrossJoinLeftOperatorFactory(int32_t id, int32_t plan_node_id, const RowDescriptor& row_descriptor,
                                  const RowDescriptor& left_row_desc, const RowDescriptor& right_row_desc,
+                                 const std::string& sql_join_conjuncts, std::vector<ExprContext*>&& join_conjuncts,
                                  std::vector<ExprContext*>&& conjunct_ctxs,
-                                 std::shared_ptr<CrossJoinContext>&& cross_join_context)
+                                 std::shared_ptr<CrossJoinContext>&& cross_join_context, TJoinOp::type join_op)
             : OperatorWithDependencyFactory(id, "cross_join_left", plan_node_id),
+              _join_op(join_op),
               _row_descriptor(row_descriptor),
               _left_row_desc(left_row_desc),
               _right_row_desc(right_row_desc),
+              _sql_join_conjuncts(sql_join_conjuncts),
+              _join_conjuncts(std::move(join_conjuncts)),
```

Comment:
Will cross join handle `_join_conjuncts `?  Differences between cross join and nestloop join: nest loop join will handle non-equal conjuncts but cross join not handle conjuncts? am I right?

Reply:
The difference between `CrossJoinOperator` and `NestloopJoinOperator`:
1. `CrossJoinOperator` could only handle `a cross join b where xxx`, it does not differentiate `join_conjuncts` and `where_conjuncts`. As a result, it could not handle `a left join b on a.id = b.id where a.name < b.name`
2. `NestloopJoinOperator` is a general operator which could handle all join types including `inner/outer/semi/anti`. The algorithm is too simple to explain

The difference between `CROSS JOIN` and `NESTLOOP JOIN`
1. The first is a logical join type, which means cartesian product
2. The nestloop join is a physical join algorithm, which could handle all kinds of join types

So, we would remove the `CrossJoinOperator` in planner and executor later, and keeps only the logical join type `cross join`.